services/index/src/api.js

```diff
@@ -66,6 +66,37 @@ builder.declare({
   return res.reply(helpers.taskUtils.serialize(task));
 });
 
+/** List tasks from given task labels */
+builder.declare({
+  method: 'post',
+  route: '/tasks/indexes',
+  query: paginateResults.query,
+  name: 'findTasksAtIndex',
+  scopes: 'index:list-tasks',
+  input: 'list-tasks-at-index.yml',
+  stability: APIBuilder.stability.experimental,
+  category: 'Index Service',
+  output: 'list-tasks-response.yml',
+  title: 'Find tasks at indexes',
+  description: [
+    'List the tasks given their labels. XXX description',
+  ].join('\n'),
+}, async function (req, res) {
+  // XXX what about authorization?
+  const indexes = req.body.indexes;
+  // XXX where does the validation happen?
```

Comment:
This is the `input: 'list-tasks-at-index.yml` (maybe add a `-request` to the name?) part of the route definition. All input will be validated automatically and rejected if validation fails